academy/0.0-B9lab-Blockchains/1_blockchain.md

```diff
@@ -0,0 +1,218 @@
+
+Other participants recognise this unlikely combination - unlikely because of the winning lottery ticket, the nonce. This unlikely combination is evidence of considerable computational work because brute force is the only known method of finding it.
+
+The network accepts the block as a de facto correct opinion about the order of transactions. There are, of course, further details such as the method of dispute resolution. This overview will suffice for now.
+
+The process disambiguates the order of the transactions even though well-meaning nodes independently arrive at slightly different opinions about the matter. The process does so **without reliance on an authoritative time source**.
+
+**A well-ordered set of blocks that each contain well-ordered transactions is a well-ordered set of all transactions that have ever occurred.**
+
+Cryptographic hash functions are instrumental in that they empower all participants to ensure that they possess an undistorted history of everything. Since all nodes can verify the chain independently, they can proceed on the assumption that all other nodes will eventually come into agreement about the history of everything. This is known as **eventual consensus**.
+
+What are the implications of blockchain's way of ordering transactions and blocks on its overall state?
+
+A blockchain starts with a known state. This is a simple matter of an initialized universe in which nothing has happened. It is often referred to as the genesis block.
+
+It proceeds by constructing a verifiable and widely agreed history of everything that has ever happened on an append-only basis. Nodes independently construct a present state of the universe by reviewing the ordered history of every change (i.e. the transactions) that has ever occurred. The history of everything that has ever occurred moves forward in time as "lottery winners" announce new transaction blocks and these are accepted as valid by a consensus of network participants.
+
+Thus, transactions being included in blocks and those becoming part of the chain leads to the state changing and being amended.
+
+## Deterministic, Atomic Operations - All or Nothing
+
+In computer science, an event is said to be "Atomic" if it can't be split into smaller parts. For example, the statement ```x = y``` is atomic if the language guarantees that ```y``` cannot be partially copied to ```x```. 
+
+In the world of databases atomicity is often specified by the developer by grouping multiple operations in a wrapper such as a COMMIT and ROLLBACK block to ensure that all of the steps complete or none of the steps execute at all. This method is often used to ensure database integrity. 
```

Comment:
Should COMMIT and ROLLBACK be in code tags (``)?

Reply:
Not exactly. I think I'll italicize. 

Syntax depends on your database implementation, e.g. MS-SQL, MySQL, Oracle. The _concept_ is to enforce atomicity at the database level.